sklearn/neighbors/dist_metrics.pyx

```diff
@@ -1093,22 +1093,29 @@ cdef class PyFuncDistance(DistanceMetric):
         self.func = func
         self.kwargs = kwargs
 
+    # in cython < 0.26, GIL was required to be acquired during definition of
```

Comment:
There has been some cleanup regarding inconsistencies between nogil and withgil on declaration vs. definition in previous versions of Cython. This seems like a fine workaround to support both versions.